content/en/blog/2025/infernece-extension-support/index.md

```diff
@@ -0,0 +1,155 @@
+
+The extension introduces two key Custom Resource Definitions that work together to address the routing challenges we've outlined. The InferenceModel resource provides an abstraction for AI-Inference workload owners to define logical model endpoints, while the InferencePool resource gives platform operators the tools to manage backend infrastructure with AI workload awareness.
+
+By extending the familiar Gateway API model rather than creating an entirely new paradigm, the inference extension enables organizations to leverage their existing Kubernetes expertise while gaining the specialized capabilities that AI workloads demand. This approach ensures that teams can adopt intelligent inference routing aligned with familiar networking knowledge and tooling.
+
+Note: InferenceModel is likely to change in future Gateway API Inference Extension releases.
+
+### **InferenceModel**
+
+The InferenceModel resource allows inference workload owners to define logical model endpoints that abstract the complexities of backend deployment.
+
+{{< text yaml >}}
+apiVersion: inference.networking.x-k8s.io/v1alpha2
+kind: InferenceModel
+metadata:
+  name: customer-support-bot
+  namespace: ai-workloads
+spec:
+  modelName: customer-support
+  criticality: Critical
+  poolRef:
+    name: llama-pool
+  targetModels:
+    - name: llama-3-8b-customer-v1
+      weight: 80
+    - name: llama-3-8b-customer-v2
+      weight: 20
+{{< /text >}}
+
+This configuration exposes a customer-support model that intelligently routes between two backend variants, enabling safe rollouts of new model versions while maintaining service availability.
+
+### **InferencePool**
+
+The InferencePool acts as a specialized backend service that understands AI workload characteristics:
+
+{{< text yaml >}}
+apiVersion: inference.networking.x-k8s.io/v1alpha2
+kind: InferencePool
+metadata:
+  name: llama-pool
+  namespace: ai-workloads
+spec:
+  targetPortNumber: 8000
+  selector:
+    app: llama-server
+    version: v1
+  extensionRef:
+    name: llama-endpoint-picker
+{{< /text >}}
+
+When integrated with Istio, this pool automatically discovers model servers through the istio’s service discovery.
+
+## **How Inference Routing Works in Istio**
+
+Istio's implementation builds on the service mesh's proven traffic management foundation. When a request enters the mesh through an Istio Gateway, it follows the standard Gateway API HTTPRoute matching rules. However, instead of using traditional load balancing algorithms, the backend is picked by an Endpoint Picker (EPP) service.
```

Comment:
"Istio Gateway" implies the [legacy Gateway type](https://istio.io/latest/docs/reference/config/networking/gateway/), but I assume you have to go through a Kubernetes-type Gateway

(Docs to make this suck less are coming soon)

Reply:
Thanks! changed to Kubernetes Gateway